Sources/VLCMediaPlayer.m

```diff
@@ -1328,6 +1360,34 @@ - (void *)libVLCMediaPlayer
     return _playerInstance;
 }
 
+- (BOOL)startRecordingAtPath:(NSString *)path
+{
+    return libvlc_media_player_record(_playerInstance, YES, [path UTF8String]);
+}
+
+- (BOOL)stopRecording
+{
+    return libvlc_media_player_record(_playerInstance, NO, nil);
+}
+
+- (BOOL)muxSubtitleFile:(NSString *)srtPath toMp4File:(NSString *)mp4Path outputPath:(NSString *)outputPath
+{
+    //check if mp4 file is valid, srt is valid
```

Comment:
I tried the following but somehow width and height directly didn't work
```suggestion
        if (![[NSFileManager defaultManager] fileExistsAtPath:srtPath] || ![[NSFileManager defaultManager] fileExistsAtPath:mp4Path]) {
        VKLog(@"no file at either srt or mp4 filepath");
        return NO;
    }
     libvlc_media_t* p_media = libvlc_media_new_path([_privateLibrary instance], [mp4Path UTF8String]);
    NSInteger width = 0;
    NSInteger height = 0;
    for (int i = 0; i < _media.tracksInformation.count; i++) {
        width = [_media.tracksInformation[i] integerForKey:VLCMediaTracksInformationVideoWidth];
        height = [_media.tracksInformation[i] integerForKey:VLCMediaTracksInformationVideoHeight];
    }

 

+    NSString *option = [NSString stringWithFormat:@":sout=#transcode{vcodec=h264,width=%ld,height=%ld,venc=avcodec{codec=h264_videotoolbox},acod(long)ec=mpga,ab=128,channels=2,samplerate=44100,soverlay}:file{dst='%@',mux=mp4}", width, height, outputPath];
```

Reply:
Are you use that tracksInformation is filled with content at this point? Additionally, this will lead to issues if the MP4 file being used to receive the subtitles does not match the media that is currently referenced, which will be the case quite often.
Further, your loop will fail if there is more than 1 video track in the given media (namely any further audio, video or SPU track).

Reply:
yes we already removed it because if you remove width and height it uses the default values of the stream